plastron-web/src/plastron/web/activitystream.py

```diff
@@ -0,0 +1,84 @@
+import json
+import logging
+from flask import Blueprint, request, jsonify, Response
+from rdflib import Graph, Namespace
+from typing import List
+from uuid import uuid4
+
+from plastron.cli.commands.publish import Command as PublishCommand
+from plastron.cli.commands.unpublish import Command as UnpublishCommand
+from plastron.namespaces import get_manager, rdf, umdact
+
+logger = logging.getLogger(__name__)
+
+activitystream = Blueprint('activities', __name__, template_folder='templates')
+
+activitystreams = Namespace('https://www.w3.org/ns/activitystreams#')
+
+@activitystream.route('/inbox', methods=['POST'])
+def new_activity():
+    try:
+        activity = Activity(from_json=request.get_json())
+        args = {
+            'uris': activity.objects,
+            'force_hidden': activity.force_hidden,
+            'force_visible': False
+        }
+        if activity.publish:
+            PublishCommand(args)
+        elif activity.unpublish:
+            UnpublishCommand(args)
+        return Response(status=201)
+    except Exception as e:
+        logger.error(f'Exception: {e}')
+        return jsonify({'error': str(e)}), 400
+
+class Activity:
```

Comment:
Take a look at this project https://pypi.org/project/activitypubdantic/ for an off-the-shelf ActivityStreams parser/validator. Note that I haven't used this at all, but it looks to be modern and under active development.

Reply:
I looked into activitypubdantic library, but the validation did not seem to cover all the cases. So, I did not use this library.